Error when cloning observation

When cloning an observation, the server throws: 
```
[ERROR] lucuma-odb - Error computing GraphQL response. 
skunk.exception.SkunkException: 
🔥  
🔥  Skunk encountered a problem related to use of begin
🔥    at «skunk internal»:0
🔥  
🔥    Problem: Nested transactions are not allowed.
🔥       Hint: You must roll back or commit the current transaction before starting
🔥             a new one.
🔥  

skunk.exception.SkunkException: Nested transactions are not allowed.
	at skunk.Transaction$.assertIdle$1$$anonfun$1(Transaction.scala:107)
	at cats.data.Kleisli.flatMap$$anonfun$2$$anonfun$1(Kleisli.scala:75)
	at cats.effect.IOFiber.succeeded(IOFiber.scala:1177)
	at cats.effect.IOFiber.runLoop(IOFiber.scala:247)
	at cats.effect.IOFiber.asyncContinueSuccessfulR(IOFiber.scala:1330)
	at cats.effect.IOFiber.run(IOFiber.scala:119)
	at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:485)
```

Invoked mutation:
```graphql
mutation ($input: CloneObservationInput!){
  cloneObservation(input: $input) {
        newObservation 
        {
          id
          title
          subtitle
          status
          activeStatus
          visualizationTime
          posAngleConstraint 
        {
          mode
          angle 
        {
          microarcseconds
        }
      
        }
      
          plannedTime {
            execution 
        {
          microseconds
        }
      
          }
          targetEnvironment {
            asterism {
              id
            }
          }
          constraintSet 
        {
          cloudExtinction
          imageQuality
          skyBackground
          waterVapor          elevationRange {
            airMass {
              min
              max
            }
            hourAngle {
              minHours
              maxHours
            }
          }
        }
      
          scienceRequirements {
            spectroscopy {
              wavelength 
        {
          picometers
        }
      
            }
          }
          observingMode 
        {
          gmosNorthLongSlit {
            grating
            filter
            fpu
            centralWavelength 
        {
          picometers
        }
      
          }
          gmosSouthLongSlit {
            grating
            filter
            fpu
            centralWavelength 
        {
          picometers
        }
      
          }
        }
      
        }
      
          }
        }
```
```json
{
  "input" : {
    "observationId" : "o-102"
    "SET" : {
      "targetEnvironment" : {
        "asterism" : [
          "t-103"
        ]
      }
    }
  }
}
```

(Sorry for the weird indentation)

The error has changed, but the problem remains. The clone is successful, but the request returns an error and this shows up in the log:
 

> [WARN ] lucuma-odb - u-10c/Todd Burnside: Internal error: Throwable: Expected single value for field 'id' of type Observation at List(newObservation, cloneObservation), found many 
> [ERROR] lucuma-odb - Error computing GraphQL response. 
> java.lang.Throwable: Expected single value for field 'id' of type Observation at List(newObservation, cloneObservation), found many

--- a/modules/service/src/main/scala/lucuma/odb/graphql/mapping/CloneObservationResultMapping.scala
+++ b/modules/service/src/main/scala/lucuma/odb/graphql/mapping/CloneObservationResultMapping.scala
@@ -12,7 +12,8 @@ trait CloneObservationResultMapping[F[_]] extends ResultMapping[F] with Observat
     ObjectMapping(
       tpe = CloneObservationResultType,
       fieldMappings = List(
-        SqlField("synthetic-id", ObservationPairsView.Left, key = true, hidden = true),
+        SqlField("synthetic-id-1", ObservationPairsView.Left, key = true, hidden = true),
+        SqlField("synthetic-id-2", ObservationPairsView.Right, key = true, hidden = true),
         SqlObject("originalObservation", Join(ObservationPairsView.Left, ObservationView.Id)),
         SqlObject("newObservation", Join(ObservationPairsView.Right, ObservationView.Id)),
       )

--- a/modules/service/src/test/scala/lucuma/odb/graphql/issue/github/359.scala
+++ b/modules/service/src/test/scala/lucuma/odb/graphql/issue/github/359.scala
@@ -1,0 +1,106 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+package issue.github
+
+import io.circe.syntax.*
+
+// https://github.com/gemini-hlsw/lucuma-odb/issues/359
+class GitHub_359 extends OdbSuite {
+  val pi = TestUsers.Standard.pi(nextId, nextId)
+  val validUsers = List(pi)
+
+  test("selecting originalObservation's id and asterism (after clone) should work") {
+    createProgramAs(pi).flatMap { pid =>
+      createObservationAs(pi, pid).flatMap { oid =>
+        query(
+          user = pi,
+          query = s"""
+            mutation {
+              cloneObservation(
+                input: {
+                  observationId: ${oid.asJson}
+                }
+              ) {
+                originalObservation {
+                  id
+                  targetEnvironment {
+                    asterism {
+                      id
+                    }
+                  }
+                }
+              }
+            }
+          """
+        )
+      }
+    }
+  }
+
+  test("selecting both observations' id and asterism (after clone) should work") {
+    createProgramAs(pi).flatMap { pid =>
+      createObservationAs(pi, pid).flatMap { oid =>
+        query(
+          user = pi,
+          query = s"""
+            mutation {
+              cloneObservation(
+                input: {
+                  observationId: ${oid.asJson}
+                }
+              ) {
+                originalObservation {
+                  id
+                  targetEnvironment {
+                    asterism {
+                      id
+                    }
+                  }
+                }
+                newObservation {
+                  id
+                  targetEnvironment {
+                    asterism {
+                      id
+                    }
+                  }
+                }
+              }
+            }
+          """
+        )
+      }
+    }
+  }
+
+  test("selecting newObservation's id and asterism (after clone) should work") {
+    createProgramAs(pi).flatMap { pid =>
+      createObservationAs(pi, pid).flatMap { oid =>
+        query(
+          user = pi,
+          query = s"""
+            mutation {
+              cloneObservation(
+                input: {
+                  observationId: ${oid.asJson}
+                }
+              ) {
+                newObservation {
+                  id
+                  targetEnvironment {
+                    asterism {
+                      id
+                    }
+                  }
+                }
+              }
+            }
+          """
+        )
+      }
+    }
+  }
+
+}

